R/rank_list.R

```diff
@@ -152,11 +154,21 @@ dropNulls <- function(x) {
 #'   )
 #'   shiny::runApp(app)
 #' }
-update_rank_list <- function(css_id, text = NULL,
+
+update_rank_list <- function(css_id,
+                             text = NULL,
+                             labels = NULL,
                              session = shiny::getDefaultReactiveDomain()) {
-  inputId <- paste0("rank-list-", css_id)
-  message <- dropNulls(list(id = inputId, text = text))
-  session$sendInputMessage(inputId, message)
+  id <- session$ns(css_id)
+  if (!is.null(text)) {
+    inputId <- paste0("rank-list-", id)
+    message <- dropNulls(list(id = inputId, text = text))
+    session$sendInputMessage(inputId, message)
+  }
+  if (!is.null(labels)) {
+    session$sendCustomMessage(type = "update-labels",
+                              message = list(inputId = id, labels = labels))
+  }
```

Comment:
I'd like to keep it within the JS binding, rather than the opposite where we have a message that points to the ID. 

Instead, can we add `labels` to `message` before we drop the `NULL`s and have the `message` object be sent to via `$sendInputMessage()`?

The corresponding `data` object in `bucketlistBinding` in `rank_list_binding.js` will need to be updated accordingly.  Something like

```js
setValue: function(el, value) {
  if (data.header) { 
    $(el).find(".bucket-list-header").text(data.header);
  }
  if (data.labels) {
    # update labels here
    # Ok to use Shiny.setInputValue here, too!
  }
},
receiveMessage: (el, data) {
  this.setValue(el, data)
}
```

When setting the labels, can the ID be inferred from the element (if needed)?

By using `setValue()`, `{shinytest2}` gains support.

Reply:
Hey @schloerke, 
I actually tried this as the first implementation, but upon testing it, the labels object didn't show up and I wasn't sure why. I can try it again just in case I messed something up and will report back shortly

Reply:
Hi @schloerke,
Thanks for the guidance on this one! I was able to get it figured out with this pseudo-code!

Reply:
🎊